db/src/models/challenge.rs

```diff
@@ -85,8 +89,21 @@ impl NewChallenge {
                 error: String::new(),
             });
         };
+        let mut nanoid = String::new();
+        loop {
```

Comment:
This is good, but we might want to move this into the DB as a trigger in the future. Then the DB can be in charge of the ids and doesn't have to send results back and forth to check for a unique id.

Reply:
Yeah that's a good idea 👍 but for not it's only during challenge creation so I decided this is the way quicker version. Changed it to an exist as well, which should be even faster.